src/sentry/notifications/helpers.py

```diff
@@ -810,19 +811,132 @@ def get_setting_options_for_recipient(
+    {user -> {notification type -> setting}}.
+
+    Args:
+        user_ids: The user ids to get notification settings for.
+        project: The project to get notification settings for.
+        organization: The organization to get notification settings for.
+    """
+    query = get_query(project=project, organization=organization, user_ids=user_ids)
+    notification_settings = NotificationSettingOption.objects.filter(query)
+
+    user_to_setting = defaultdict(dict)
+    for ns in notification_settings:
+        if ns.scope_type == NotificationScopeEnum.PROJECT.value:
+            user_to_setting[ns.user][ns.type] = ns
+
+    for ns in notification_settings:
+        if ns.scope_type == NotificationScopeEnum.ORGANIZATION.value:
+            ns_dict = user_to_setting[ns.user]
+            if ns.type not in ns_dict:
+                user_to_setting[ns.user][ns.type] = ns
+
+    for ns in notification_settings:
+        if ns.scope_type == NotificationScopeEnum.USER.value:
+            ns_dict = user_to_setting[ns.user]
+            if ns.type not in ns_dict:
+                user_to_setting[ns.user][ns.type] = ns
+
+    defaults = get_type_defaults()
+    for _, ns_dict in user_to_setting.items():
+        for type in NotificationSettingEnum:
+            if type not in ns_dict and type in defaults:
+                user_to_setting[ns.user][type] = defaults[type]
+
+    return user_to_setting
+
+
+def get_setting_providers_for_users(
+    user_ids: Iterable[int],
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> MutableMapping[RpcActor, MutableMapping[NotificationSettingEnum, NotificationSettingProvider]]:
+    """
+    Returns a map of users to NotificationSettingProvider by provider:
+    {user -> {provider -> setting}}.
+
+    Args:
+        user_ids: The user ids to get notification settings for.
+        project: The project to get notification settings for.
+        organization: The organization to get notification settings for.
+    """
+    query = get_query(project=project, organization=organization, user_ids=user_ids)
+    notification_settings = NotificationSettingProvider.objects.filter(query)
+
+    user_to_setting = defaultdict(dict)
+    for ns in notification_settings:
+        if ns.scope_type == NotificationScopeEnum.PROJECT.value:
+            user_to_setting[ns.user][ns.provider] = ns
+
+    for ns in notification_settings:
+        if ns.scope_type == NotificationScopeEnum.ORGANIZATION.value:
+            ns_dict = user_to_setting[ns.user]
+            if ns.provider not in ns_dict:
+                ns_dict[ns.provider] = ns
+
+    for ns in notification_settings:
+        if ns.scope_type == NotificationScopeEnum.USER.value:
+            ns_dict = user_to_setting[ns.user]
+            if ns.provider not in ns_dict:
+                ns_dict[ns.provider] = ns
+
+    # TODO: Are there any provider defaults to fill in?
```

Comment:
Is there a notion of default providers for notifications if one isn't explicitly set?

Reply:
@snigdhas Yea, see `src/sentry/api/endpoints/notification_defaults.py`